src/common/com/intellij/plugins/haxe/model/HaxeFunctionModelUtils.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2015 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model;
+
+import com.intellij.plugins.haxe.lang.psi.HaxeTypeTag;
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.Nullable;
+
+public class HaxeFunctionModelUtils {
+  @Nullable
+  static public PsiElement getBodyPsi(PsiElement methodElement) {
+    PsiElement[] children = methodElement.getChildren();
+    if (children.length == 0) return null;
+    PsiElement child = children[children.length - 1];
+    if (child instanceof HaxeTypeTag) return null;
```

Comment:
This breaks when the function does not have a code block, but merely contains an expression.  Against the following code, the return value of getBodyPsi is the PsiToken for the terminating semi-colon
```haxe
   public function test()
      (1);
```
And, on a further note, it looks like we have a parsing error because the following code parses as a function with a return statement, followed by a variable declaration, though it compiles just fine on try.haxe.org.
```haxe
    public function test2(size:Int)
        return var a : Array<Int> = [for (i in 0...size) i];
```
In which case, the body of this function returns as the "return" keyword, which I'm sure breaks the typing code. :/  I'll file the issue.